### ISSUE TITLE
Spyder 4.0.1 hangs during editing code

### ISSUE THREAD

**Issue by eafpres** [spyder-ide/spyder#11559]
***Problem Description

When editing code, say a Python source file with ~600 lines, sometimes the entire application becomes non-responsive.  Checking task manager, it stops using CPU and I/O, it's just dead.

To recover, I have to terminate from task manager or close the miniconda console.

On restart, usually there is an option to restore.  However, yesterday after working for at least 30 minutes, it hung, and the code I had been working on was not restored (perhaps if never saved, it's never autosaved?)

### What steps reproduce the problem?

This is the problem, I have found no way to repro so far.  I wonder if there is a log or anything I could provide to you that would be helpful?  If I ever get a repro I'll of course share it.

### What is the expected output? What do you see instead?

1) Application does not hang and force killing the process to restart
2) Autosave un-saved (new) source code for recovery

### Paste Traceback/Error Below (if applicable)

```
none--it just hangs

```

## Versions

Spyder 4.0.1 
Python 3.7.6 64-bit
Qt 5.9.6
PyQt5 5.9.2
Windows 10 

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.12.0 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.8.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.7 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.8 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.3 (OK)
sphinx >=0.6.6               :  2.3.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  0.29.13 (OK)
matplotlib >=2.0.0           :  2.2.3 (OK)
numpy >=1.7                  :  1.17.4 (OK)
pandas >=0.13.1              :  0.25.1+0.g171c716.dirty (OK)
scipy >=0.17.0               :  1.3.2 (OK)
sympy >=0.7.3                :  1.5.1 (OK)

```

**Comment by gitpickle** [comment]
I've had similar problems in other editors, such as Toad.  There probably
some series of content within your file that is causing the issue and the
root cause needs to be determined.  You may need submit the entire source
file.

**Comment by goanpeca** [comment]
@eafpres you could run spyder from the source with 

```
python bootstrap --debug
```

And share with us the logs once you find it blocks. Thanks!

**Comment by ccordoba12** [comment]
Duplicate of #11222

**Comment by ccordoba12** [comment]
This should be fixed once we release 4.1.0 in a couple of weeks. If not, please open a new issue again.

**Comment by eafpres** [comment]
OK, thanks.  I just asked this question in 11222, which you can ignore.
